src/Build/Construction/ProjectTargetElement.cs

```diff
@@ -58,22 +58,22 @@ private ProjectTargetElement(XmlElementWithLocation xmlElement, ProjectRootEleme
         /// <summary>
         /// Get an enumerator over any child item groups
         /// </summary>
-        public ICollection<ProjectItemGroupElement> ItemGroups => new ReadOnlyCollection<ProjectItemGroupElement>(Children.OfType<ProjectItemGroupElement>());
+        public ICollection<ProjectItemGroupElement> ItemGroups => GetChildrenOfType<ProjectItemGroupElement>();
```

Comment:
Maybe I'm wrong here, but I guess all of the changes that cast `GetChildrenOfType` to `ICollection` do nothing here, because you cast you collection to `ICollection` and so the consumer of this code iterating through `ICollection` interface won't call a special `GetEnumerator` method that returns your struct, but rather will call a method on the interface that still would box the iterator.

Reply:
Bah, I think you're right. This would only save on the OfTypeIterators. Maybe still substantial enough to be worth it, but it's less compelling now. Might need to actually compare and gather data.

Reply:
Gotcha. I'm not sure we can make it a known public type very easily though since the existing impl is lazy. It's (effectively) a linked list and while the enumerator may box, it's not knowable whether allocating the full collection would be beneficial or not.

Consider a naive consumer who just enumerates. With these changes, it would allocate the collection (just a wrapper) and box the enumerator. If we returned a well-known collection (let's say List<T>), we'd have to enumerate the full collection and populate the whole collection, which takes more memory due to the underlying storage (array).